components/bsx/Gallery/Item/AvailableActions.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+  <div>
+    <Loader v-model="isLoading" :status="status" />
+    <ActionList v-if="accountId" :actions="actions" @click="handleAction" />
+    <component
+      class="mb-4"
+      v-if="showMeta"
+      :is="showMeta"
+      @input="updateMeta"
+      emptyOnError />
+    <SubmitButton v-if="showSubmit" @click="submit">
+      {{ $t('nft.action.submit', [selectedAction]) }}
+    </SubmitButton>
+  </div>
+</template>
+
+<script lang="ts">
+import { NFTAction } from '@/components/unique/NftUtils'
+import nftById from '@/queries/nftById.graphql'
+import { bsxParamResolver, getApiCall } from '@/utils/gallery/abstractCalls'
+import { notificationTypes, showNotification } from '@/utils/notification'
+import { unpin } from '@/utils/proxy'
+import {
+  actionComponent,
+  getActionList,
+  iconResolver,
+} from '@/utils/shoppingActions'
+import shouldUpdate from '@/utils/shouldUpdate'
+import Connector from '@kodadot1/sub-api'
+import { Component, mixins, Prop } from 'nuxt-property-decorator'
+import { createTokenId } from '~/components/unique/utils'
+import { isSameAccount } from '~/utils/account'
+import AuthMixin from '~/utils/mixins/authMixin'
+import KeyboardEventsMixin from '~/utils/mixins/keyboardEventsMixin'
+import MetaTransactionMixin from '~/utils/mixins/metaMixin'
+import PrefixMixin from '~/utils/mixins/prefixMixin'
+
+const components = {
+  ActionList: () => import('@/components/rmrk/Gallery/Item/ActionList.vue'),
+  AddressInput: () => import('@/components/shared/AddressInput.vue'),
+  BalanceInput: () => import('@/components/shared/BalanceInput.vue'),
+  SubmitButton: () => import('@/components/base/SubmitButton.vue'),
+  Loader: () => import('@/components/shared/Loader.vue'),
+}
+
+@Component({ components })
+export default class AvailableActions extends mixins(
+  PrefixMixin,
+  KeyboardEventsMixin,
+  MetaTransactionMixin,
+  AuthMixin
+) {
+  @Prop(String) public currentOwnerId!: string
+  @Prop() public price!: string
+  @Prop() public nftId!: string
+  @Prop(String) public collectionId!: string
+  @Prop({ type: Array, default: () => [] }) public ipfsHashes!: string[]
+
+  private selectedAction: NFTAction | '' = ''
+  private meta: string | number = ''
+
+  get actions() {
+    return getActionList('bsx', this.isOwner, this.isAvailableToBuy)
+  }
+
+  get isOwner(): boolean {
+    this.$consola.log(
+      '{ currentOwnerId, accountId }',
+      this.currentOwnerId,
+      this.accountId
+    )
+
+    return Boolean(
+      this.currentOwnerId &&
+        this.accountId &&
+        isSameAccount(this.currentOwnerId, this.accountId)
+    )
+  }
+
+  get showSubmit() {
+    return this.selectedAction && (!this.showMeta || this.metaValid)
+  }
+
+  get metaValid() {
+    if (typeof this.meta === 'number') {
```

Comment:
if `this.meta` would be int, maybe use `Number.isInteger` could be better.

I was always felt insecure about `typeof`